app/v2/services/JsonMessageParsingService.scala

```diff
@@ -59,11 +59,7 @@ class JsonMessageParsingServiceImpl @Inject() (implicit materializer: Materializ
         .map {
           mt =>
             val root = mt.replace("\"", "")
-            MessageType.messageTypesSentByDepartureTrader
-              .filterNot(
-                messageType => messageType == MessageType.DeclarationData
-              )
-              .find(_.rootNode == root) match {
+            MessageType.updateMessageTypesSentByDepartureTrader.find(_.rootNode == root) match {
```

Comment:
@sonisn1988  It looks like we still have it as the responsibility of the json/xml parsers to determine whether the message type is valid. It seems to go against what @drnaylor-oc suggested as it checks that the message type fits the expected list rather than just identifying a message type (from a list of all message types?) and checking if it's the expected message type in the controller. I think Dan's suggestion would mean that we just return the value in the StartElement/Json root node and return that to the controller. The controller would then check if it fits the expected message types for the movement. Perhaps we could do this as a tech debt to prevent not completing this prior sprint end?

Reply:
Rather, it's not the responsibility of this service to determine which list of message types to check against. I'd argue that the Seq of messages it should be able to get should be supplied to this service, this shouldn't determine it itself.

I'm basically suggesting that `MessageType.updateMessageTypesSentByDepartureTrader` is really something that should be passed in as an argument to the function, i.e. the controller determines the valid values, and this service uses that list to find if that's the case.

However, we're going to end up going around in circles on this one, as this isn't the Json ticket but the XML one, let's just get this done here.

Reply:
Thanks @drnaylor-oc / @liandra-giacomine I did the changes for both xml and json as suggested above. Please see below with this commit
https://github.com/hmrc/common-transit-convention-traders/pull/347/commits/91d9905f58d45b651237de07a9a3b298d55d62dd